### PR TITLE
Convert GraylogProvider to Kotlin

### PR DESCRIPTION
**Pull request by jahunt1**
This one was fun since there are more classes to be converted.

### CODE REVIEW

File: graylog-provider/src/main/java/com/busybusy/graylog_provider/EventJsonizer.kt [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+        append("[")
+        for (element in attributeList) {
+            if (element is Map<*, *>) // recursive case
+            {
+                append(getJsonFromMapRecursive(element as Map<String, Any>)).append(", ")
+            } else if (element is List<*>) // recursive case
+            {
+                append(getJsonFromListRecursive(element)).append(", ")
+            } else {
+                if (element is String) // gotta use escape quotes for JSON strings
```
Comment (simon-the-canadian) [inline graylog-provider/src/main/java/com/busybusy/graylog_provider/EventJsonizer.kt:175]: would it work to pull this nested if/else if/else statement out to be on the same level?

**Review by simon-the-canadian — APPROVED** [review]
Looks great, just a few nit picky things.

File: graylog-provider/src/main/java/com/busybusy/graylog_provider/EventJsonizer.kt [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+    private fun underscorePrefix(fieldName: String): String =
+            if (fieldName.startsWith("_")) "" else "_"
+
+    @Suppress("UNCHECKED_CAST")
+    internal fun getJsonFromMapRecursive(attributeMap: Map<String, Any>): String = buildString {
+        append("{")
+        for (innerAttribute in attributeMap.keys) {
+            val innerAttributeValue = attributeMap[innerAttribute]
+            if (innerAttributeValue is Map<*, *>) // recursive case
+            {
```
[the remark this answers is not in the document]
Comment (jahunt1) [reply graylog-provider/src/main/java/com/busybusy/graylog_provider/EventJsonizer.kt:131]: :disappointed:

File: graylog-provider/src/main/java/com/busybusy/graylog_provider/EventJsonizer.kt (hunk 1 above)
Comment (jahunt1) [reply graylog-provider/src/main/java/com/busybusy/graylog_provider/EventJsonizer.kt:175]: Yes! In fact, when we put them all on the same level we can replace the whole thing with a nice `when`!

**Pull request merged**